pkg/builddocker/build.go

```diff
@@ -44,6 +45,82 @@ func GenerateDockerfile(w io.Writer, env hcl2nix.OCIArtifact, platform string) e
 		return err
 	}
 
+	return nil
+
+}
+
+// ModifyDockerfile modifies the Dockerfile with the specified tag
+func ModifyDockerfile(path, tag string, dev bool) error {
+	var dockerfilePath string
+	if path != "" {
+		dockerfilePath = path + "/Dockerfile"
+	} else {
+		dockerfilePath = "./Dockerfile"
+	}
+
+	if _, err := os.Stat(dockerfilePath); os.IsNotExist(err) {
+		return fmt.Errorf("dockerfile not found")
+	}
+
+	file, err := os.Open(dockerfilePath)
+	if err != nil {
+		return fmt.Errorf("error opening Dockerfile: %v", err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	lines := []string{}
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
+	if err := scanner.Err(); err != nil {
+		return fmt.Errorf("error reading Dockerfile: %v", err)
+	}
+
+	var searchTag string
```

Comment:
Can we add a small unit test for the rest to make sure the swap is functioning correctly? 
I have a feeling we'll need to modify this in future and having unit test will make sure we don't break something accidentally in future

Reply:
added